### PR TITLE
add Build.TestOccurrences

### PR DESCRIPTION
**Pull request by nex-54**
### Issue details

Add Build.TestOccurrences property

### Checklist
- [x] All unit tests passed on build server
- [x] Acceptance test(s) covers new/modified functionality 
- [x] Code coverage is at least the same or higher
- [x] Breaking change in public API

var build = connectedTc.GetBuild(123);
var failedTests = build.TestOccurrences.Failed

### CODE REVIEW

File: FluentTc.Tests/AcceptanceTests.cs [hunk 1]
```diff
@@ -108,8 +108,46 @@ public void GetBuildFullResponse_Id_Build()
...
+                    }
+                });
+
+            var connectedTc = new RemoteTc().Connect(_ => _.AsGuest(), teamCityCaller);
+
+            // Act
+            var build = connectedTc.GetBuild(123);
+
+            // Assert
+            build.TestOccurrences.Should().NotBe(null);
```
Comment (borismod) [inline FluentTc.Tests/AcceptanceTests.cs]: There is a dedicated method on FluentAssertions to validate object for not being null

```C#
obj.Should().NotBeNull();
```

File: FluentTc/Domain/Build.cs [hunk 2]
```diff
@@ -14,6 +14,7 @@ public interface IBuild
         DateTime QueuedDate { get; }
         BuildConfiguration BuildConfiguration { get; }
         Agent Agent { get; }
+        TestOccurrences TestOccurrences { get; }
```
Comment (borismod) [inline FluentTc/Domain/Build.cs]: I know that this is confusing, but FluentTc is under a refactoring process. During the process concrete classes are replaced with read-only interfaces. It means interfaces with only getters. Build class was already refactored to IBuild interface and it has only getters. This approach makes testing much easier. Can you please replace TestOccurences class to an interface?

File: FluentTc/Domain/TestOccurrences.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+namespace FluentTc.Domain
+{
+    public class TestOccurrences
```
Comment (borismod) [inline FluentTc/Domain/TestOccurrences.cs]: Can you please extract an interface from this class and make all its properties read-only, i.e. having only getters?

**Comment by borismod** [comment]
@nex-54 please make sure your test coverage is at least the same as it was before
You can examine the code coverage of your build here:
http://teamcity.codebetter.com/viewLog.html?buildId=221729&buildTypeId=FluentTc_FluentTcDevelop&tab=coverage_dotnet

I also recommend using DotCover for VisualStudio for finding the uncovered code.

**Comment by nex-54** [comment]
I will fix code coverage in all my pull requests little bit later.

**Comment by nex-54** [comment]
Done. Code coverage is at least the same or higher.

**Comment by borismod** [comment]
Thank you for your contribution, Artem!

**Pull request merged**